RealmSwift/MongoClient.swift

```diff
@@ -483,4 +502,230 @@ extension MongoCollection {
+        private let collection: MongoCollection
+        private var changeStream: ChangeStream?
+        private var subscriber: S?
+
+        init(collection: MongoCollection, subscriber: S, queue: DispatchQueue = .main) {
+            self.collection = collection
+            self.subscriber = subscriber
+            setup(queue: queue)
+        }
+
+        init(collection: MongoCollection, subscriber: S, filterIds: [ObjectId], queue: DispatchQueue = .main) {
+            self.collection = collection
+            self.subscriber = subscriber
+            setup(with: filterIds, queue: queue)
+        }
+
+        init(collection: MongoCollection, subscriber: S, matchFilter: Document, queue: DispatchQueue = .main) {
+            self.collection = collection
+            self.subscriber = subscriber
+            setup(with: matchFilter, queue: queue)
+        }
+
+        func request(_ demand: Subscribers.Demand) { }
+
+        func cancel() {
+            changeStream?.close()
+        }
+
+        private func setup(queue: DispatchQueue) {
+            changeStream = collection.watch(delegate: self, queue: queue)
+        }
+
+        private func setup(with filterIds: [ObjectId], queue: DispatchQueue) {
+            changeStream = collection.watch(filterIds: filterIds, delegate: self, queue: queue)
+        }
+
+        private func setup(with matchFilter: Document, queue: DispatchQueue) {
+            changeStream = collection.watch(matchFilter: matchFilter, delegate: self, queue: queue)
+        }
+
+        func changeStreamDidOpen(_ changeStream: RLMChangeStream) { }
+
+        func changeStreamDidClose(with error: Error?) {
+            guard let error = error else {
+                subscriber?.receive(completion: .finished)
+                return
+            }
+            subscriber?.receive(completion: .failure(error))
+        }
+
+        func changeStreamDidReceive(error: Error) {
+            subscriber?.receive(completion: .failure(error))
+        }
+
+        func changeStreamDidReceive(changeEvent: AnyBSON?) {
+            guard let changeEvent = changeEvent else {
+                return
+            }
+            _ = subscriber?.receive(changeEvent)
+        }
+    }
+
+    public struct WatchPublisher: Publisher {
+        public typealias Output = AnyBSON
+        public typealias Failure = Error
+
+        private let collection: MongoCollection
+        private let queue: DispatchQueue
+        private let filterIds: [ObjectId]?
+        private let matchFilter: Document?
+
+        init(collection: MongoCollection, queue: DispatchQueue, filterIds: [ObjectId]? = nil, matchFilter: Document? = nil) {
+            self.collection = collection
+            self.queue = queue
+            self.filterIds = filterIds
+            self.matchFilter = matchFilter
+        }
+
+        /// :nodoc:
+        public func receive<S: Subscriber>(subscriber: S) where
+            Self.Failure == S.Failure, Self.Output == S.Input {
```

Comment:
```suggestion
            Failure == S.Failure, Output == S.Input {
```

`Self.` isn't applicable here since it isn't a protocol.